### PR TITLE
Adding compiled binaries and platform scripts

### PR DESCRIPTION
**Pull request by bbaltz505**
Adding compiled binaries and JTAG scripts to 'dist' folder. these folders can be packaged up for each OS.

### CODE REVIEW

**Comment by bbaltz505** [comment]
@calvinatintel please review - especially README.debugging

**Comment by calvinatintel** [comment]
Removed `.DS_Store files`

**Comment by calvinatintel** [comment]
Renamed `README` to `README.openocd` to show that the files aren't from us

**Comment by calvinatintel** [comment]
@bbaltz505 Does building with IDE generate `arduino101_sketch.debug.elf`? I don't think I see it in `/tmp/`

**Comment by bbaltz505** [comment]
The IDE copies the debug ELF to {build.path}..
I'm not sure where that is on OSX. Can you turn on verbose mode and see?

**Comment by calvinatintel** [comment]
The file ended up here:
`cp -f "/var/folders/vj/z99xs1f94yb50z371k13h9y80000gq/T/buildea32c38e9f50eb55a19e3223c29708a6.tmp/sketch_jul14a.ino.elf" "/var/folders/vj/z99xs1f94yb50z371k13h9y80000gq/T/buildea32c38e9f50eb55a19e3223c29708a6.tmp/../arduino101_sketch.debug.elf"`

**Comment by calvinatintel** [comment]
I'm not sure we can get that path without first running the IDE to build a blank sketch

**Comment by bbaltz505** [comment]
Ok. After closing the IDE, what part of that path still exists? Usually, The build folder gets wiped out. The debug ELF should be copied to the parent folder though so that it remains. We may need a special version of the OSX copy command to copy the ELF somewhere else if it's getting wiped out.

**Comment by bbaltz505** [comment]
Also, how should OpenOCD be hosted for installation with the CODK? if we pull it in as a repo (referenced by the manifest) then we'll need to restructure it. Otherwise we jost host it as a zip file and the CODK setup script pulls it down?

**Comment by calvinatintel** [comment]
> The build folder gets wiped out. The debug ELF should be copied to the parent folder though so that it remains.

Yup, that's what happened.

> Also, how should OpenOCD be hosted for installation with the CODK?

Let me think about this for a bit. It'd be nice if it just lives inside the flashpack repo. Maybe I can use `repo` to mix them into one dir somehow.

**Comment by calvinatintel** [comment]
> We may need a special version of the OSX copy command to copy the ELF somewhere else if it's getting wiped out. 

Do you think Arduino would mind if we copy it out to say `~/arduino101.elf`?

**Comment by bbaltz505** [comment]
@calvinatintel ready for review. If approved, we can create the zip packages for downloadmirror

**Comment by bbaltz505** [comment]
@calvinatintel btw, I did think about having the ELF saved to the sketchbook.path. We can ask if Arduino would mind that. It would be very visible to the average/new user though.

**Comment by bbaltz505** [comment]
@calvinatintel I added the "expected" error message GDB displays after connecting to the target:

**Comment by bbaltz505** [comment]
@calvinatintel pls review. OSX users will need to get the actual build folder and debug sketch location from verbose build output in the IDE. I don't think we should clutter up the average user's system so i don't want to save it in the sketch folder or sketchbook.

File: dist/osx/README.debugging [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+connection.
+
+After uploading a sketch to the board, run these commands in SEPARATE windows
+1. "<openocd_dir>/bin/openocd " 
+      -f "<openocd_dir>/scripts/interface/ftdi/flyswatter2.cfg" 
+      -f "<openocd_dir>/scripts/board/quark_se.cfg"
+2. <gdb> "<tmp_dir>/arduino101_sketch.debug.elf"
+
+In GDB, run the command: 
+   target remote localhost:3333
```
Comment (calvinatintel) [inline dist/osx/README.debugging:11]: Can you please tell me the expected output of this command? I got this far but I'm not sure what to do afterwards.

**Comment by bbaltz505** [comment]
@calvinatintel  After running the target command you should get the GDB prompt back so you can begin debugging. For the Blink sketch, i normally do:
gdb> l loop    (which lists the first few lines of the loop function in the sketch)
gdb> b loop:25  (sets breakpoint on the digitalWrite line)
gdb> continue    (restarts the program, I believe)
I'll assume they know how to usre GDB but I need to note the false error message listed when the "target" command is run

**Comment by bbaltz505** [comment]
@SidLeung @bigdinotech FYI, does this README for JTAG GDB users make sense?
https://github.com/01org/OpenOCD/blob/b91eabe5d5b70813070aacfa14990a2c2d409c9c/dist/linux32/README.debugging

**Comment by calvinatintel** [comment]
Confirmed to work on OSX. Thank you!

**Comment by bbaltz505** [comment]
@calvinatintel do you think this is good to go? This PR is only for the files (dist/*) to be packaged up and installed for the Arduino IDE.

**Comment by calvinatintel** [comment]
Well, I was hoping that there is a partial download solution so that we can pick up just `dist` dir, but I wasn't able to find one. The content works. Perhaps we should pull in the whole thing with the source into boards package so that we meet the open sourcing requirements at the same time?

**Comment by bbaltz505** [comment]
@calvinatintel I added a simple README with a pointer to the GitGHub location and our build instructions from Jenkins. I'll go ahead and merge to test out the uploaded binary feature for a test release.

**Pull request merged**